### PR TITLE
Remove flutter api named and optional parameters

### PR DESCRIPTION
**Pull request by tarrinneal**
fixes https://github.com/flutter/flutter/issues/140045
replaces https://github.com/flutter/packages/pull/5663 as no longer needed.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/pigeon/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 15.0.2
+
+* Prevents optional and non-positional parameters in flutterApis.
```
Comment (stuartmorgan) [inline packages/pigeon/CHANGELOG.md:3]: Nit: Flutter APIs.

**Review by stuartmorgan — APPROVED** [review]
LGTM, but isn't this only a partial fix for that issue? It says that Dart test code for Host APIs is also incorrect.

**Comment by schultek** [comment]
Yeah it was generating incorrect code for the host api tests (which internally use the flutter api code) so I thought when fixing the bug its not much more to support flutter api in general.

**Comment by stuartmorgan** [comment]
> so I thought when fixing the bug its not much more to support flutter api in general.

Meaningfully supporting named and optional parameters in Flutter APIs would require supporting those things in all of the native generators, since that's where they are called from, and it would be a substantial amount of work to define what that would even mean in each language.

**Comment by schultek** [comment]
Its obvs your choice, but I thought it could just be named on dart side and treated as normal positionals on native side. From my tests that seemed to work.

**Comment by stuartmorgan** [comment]
> I thought it could just be named on dart side and treated as normal positionals on native side. From my tests that seemed to work.

It's not that it doesn't work, it's that it doesn't really accomplish anything. The value of optional and named parameters is to callers, not within the implementation, and Pigeon clients never call Flutter API Dart code from Dart.

**Comment by tarrinneal** [comment]
> LGTM, but isn't this only a partial fix for that issue? It says that Dart test code for Host APIs is also incorrect.

So right you are.

Sorry @schultek for wrongfully closing your pr. I've pulled your changes in to this one to make sure you get credit.

File: packages/pigeon/CHANGELOG.md (hunk 1 above)
Comment (stuartmorgan) [reply packages/pigeon/CHANGELOG.md:3]: This didn't get changed.

**Review by stuartmorgan — APPROVED** [review]
LGTM with nit.

File: packages/pigeon/CHANGELOG.md (hunk 1 above)
Comment (tarrinneal) [reply packages/pigeon/CHANGELOG.md:3]: change must have gotten lost when I merged branches, I definitely changed it.

**Pull request merged**